kusto/kql/query_parameters.go

```diff
@@ -0,0 +1,67 @@
+package kql
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type ParamVals struct {
+	paramType string
+	value     string
+}
+
+type StatementQueryParameters struct {
+	parameters map[string]ParamVals
+}
+
+var buildPool = sync.Pool{
```

Comment:
@AsafMah , do we want to keep a pool of string builders (memory management wise and such)?

Reply:
I think it's a really good idea actually.

Reply:
We should probably do it for StatementBuilder too, but probably out of scope here

Reply:
@ronmonetaMicro after talking w/ Yochai, just remove the syncpool here too.

Reply:
not sure i understand, Whats the need in a reset? Strings already have a Reset function. and i didnt understand what change you wanted in Builder.go